src/server/scripts/Northrend/zone_howling_fjord.cpp

```diff
@@ -160,12 +136,24 @@ enum Entries
     SPELL_HEALING_POTION         = 17534
 };
 
-class npc_apothecary_hanes : public CreatureScript
+struct npc_apothecary_hanes : public npc_escortAI
 {
-public:
-    npc_apothecary_hanes() : CreatureScript("npc_apothecary_hanes") { }
+    npc_apothecary_hanes(Creature* creature) : npc_escortAI(creature) { }
+    uint32 PotTimer;
 
-    bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override
+    void Reset() override
+    {
+        SetDespawnAtFar(false);
+        PotTimer = 10000; //10 sec cooldown on potion
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        if (Player* player = GetPlayerForEscort())
+            player->FailQuest(QUEST_TRAIL_OF_FIRE);
+    }
+
+    bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest)
```

Comment:
No, this can't be moved to the new system, need to use the old system for OnQuestAccept.

Reply:
Are you 100% sure? Build should fail but isnt